packages/client/hmi-client/src/composables/useCharts.ts

```diff
@@ -540,22 +540,42 @@ export function useCharts(
 	const useCompareDatasetCharts = (
 		chartSettings: ComputedRef<ChartSetting[]>,
 		selectedPlotType: ComputedRef<PlotValue>,
-		baselineIndex: ComputedRef<number>
+		// baselineIndex: ComputedRef<number>,
+		datasets: Ref<Dataset[]>,
+		modelConfigurations,
+		interventionPolicies
 	) => {
 		const compareDatasetCharts = computed(() => {
 			const charts: Record<string, VisualizationSpec> = {};
 			if (!isChartReadyToBuild.value) return charts;
 
-			// Make baseline black
-			const colorScheme = cloneDeep(CATEGORICAL_SCHEME);
-			colorScheme.splice(baselineIndex.value, 0, 'black');
+			const interventionNameColorMap: Record<string, string> = {};
+			const interventionNameScoreMap: Record<string, string> = {};
```

Comment:
I don't see why these variables are needed to be declared out side of `setInterventionColorAndScoreMaps` function and let function modifies them. Can we move them inside the function and let `setInterventionColorAndScoreMaps` return them?